### PR TITLE
Add formatting configuration options

### PR DESCRIPTION
**Pull request by JakeStanger**
Adds ability to configure the formatting of the details and state strings in Discord.

Includes support for tokens within the string to use MPD values. More details in updated readme.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -75,33 +93,87 @@ fn idle(hosts: &[String]) -> MPDClient {
...
+        "album" => song.tags.get("Album").unwrap_or(&EMPTY).clone(),
+        "artist" => song.tags.get("Artist").unwrap_or(&EMPTY).clone(),
+        "date" => song.tags.get("Date").unwrap_or(&EMPTY).clone(),
+        "disc" => song.tags.get("Disc").unwrap_or(&EMPTY).clone(),
+        "genre" => song.tags.get("Genre").unwrap_or(&EMPTY).clone(),
+        "track" => song.tags.get("Track").unwrap_or(&EMPTY).clone(),
+        "duration" => format_duration(song.tags.get("duration")).unwrap_or(String::new()),
+        _ => token.to_string(),
+    }
+}
```
Comment (JakeStanger) [inline src/main.rs]: I want to re-write this to be less disgusting before merging it

**Comment by Phate6660** [comment]
Oh wow, this is really nice. I tried to do my own version of this but failed miserably. (The mpd crate is so confusing to use, at least for me. The most I've ever been able to get it to do is toggle the music state and display very verbose status information. Plus, I've never figured out how to properly read/write from/to a config.)

I like all the options you added, as well as how easy you made it to configure the output.

A couple questions though: 

1. Do you know if it's possible to obtain the current elapsed time of the song? So for example, I could have it output something like `artist - title [0:13/4:51]`?

2. Is it possible to have more than 2 lines? I'm fairly certain I've seen other discordrpc scripts for other programs have up to like 4 or 5 lines, so I assume it's probably possible. So for example I could do something like:

```
title [elapsed/duration]
album (date)
artist
genre
```

File: src/main.rs (hunk 1 above)
Comment (Phate6660) [reply src/main.rs]: If it's any consolation, I don't think it's disgusting.

Granted, I am by no means a professional programmer. Everything I've ever done has been self-taught, so I'm sure the way I see things is much different than someone who is a professional.

**Comment by JakeStanger** [comment]
>     1. Do you know if it's possible to obtain the current elapsed time of the song? So for example, I could have it output something like `artist - title [0:13/4:51]`?

This should definitely be possible, I'll have a look into that



>     2\. Is it possible to have more than 2 lines? I'm fairly certain I've seen other discordrpc scripts for other programs have up to like 4 or 5 lines, so I assume it's probably possible. 

Do you have any examples? I had a try but didn't have any luck with that. The fields definitely don't accept newlines.

File: src/main.rs (hunk 1 above)
Comment (JakeStanger) [reply src/main.rs]: It's clear what it does, but it's a lot of repeated code and it's certainly not a very efficient approach. This is my first real program in Rust so I'm still trying to get comfortable with the basics.

If I don't get anywhere in the next couple of days, I'll merge it as-is and re-write it some other day

**Comment by Phate6660** [comment]
> This should definitely be possible, I'll have a look into that

Awesome! Thank you.

> Do you have any examples?

I do, but unfortunately it's in lua. I use a discordrpc script for `mpv` and it manages to output like this:

```
mpv Media Player
title (or filename if no title tag exists)
status
amount of time left
```
[Here's](https://github.com/cniw/mpv-discordRPC) the Github repo for now if you're interested, and in the meantime I'll try to pinpoint how exactly it does it.

**Comment by JakeStanger** [comment]
So that's the elapsed time support in. 

From looking at that repo, I think it just sets the "time left" field which Discord then natively handles (you can show an elapsed time OR a time left on the 3rd line and nothing else. I'll keep having a look in the hope's that I'm wrong\

File: src/main.rs (hunk 1 above)
Comment (JakeStanger) [reply src/main.rs]: Many thanks to @simvux for tidying this up for me!

**Comment by JakeStanger** [comment]
The code is in a state where I'm happy to merge this. I will have another poke around in the future but I've not found anything to suggest mult-line support in the fashion you suggested.

**Pull request merged**